hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/HoodieSparkTable.java

```diff
@@ -67,6 +69,12 @@ protected HoodieSparkTable(HoodieWriteConfig config, HoodieEngineContext context
 
   @Override
   protected HoodieIndex<T, JavaRDD<HoodieRecord<T>>, JavaRDD<HoodieKey>, JavaRDD<WriteStatus>> getIndex(HoodieWriteConfig config, HoodieEngineContext context) {
-    return SparkHoodieIndex.createIndex(config);
+    String persistIndexType = this.metaClient.getTableConfig().getProperties().getProperty(HoodieIndexConfig.INDEX_TYPE_PROP);
```

Comment:
this part does not need the `TableNotFoundException` checking?

Reply:
yes , because the metaclient have created